package_repo_resource.md

```diff
@@ -0,0 +1,71 @@
+---
+RFC: unassigned
+Author: Thom May <tmay@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Add a resource to manage package repositories
+
+Provide a core resource to allow a user to add a package repository to
+their system.
+
+## Motivation
+
+    As a cookbook author,
+    I want to write cookbooks that rely only on core resources,
+    so that I can minimise my dependencies.
+
+## Specification
+
+Create a new `package_repo` resource, similar to 
+```
+package_repo "foo" do
+  description "A descriptive string"
+  base_url "https://some/url"
+  gpgkey "https://some/other/url"
```

Comment:
Are the semantics around "package_repo" congruent across all the different
types?

I can't imagine they are.... this smells like a leaky abstraction to me.

-s

On Wed, Mar 4, 2015 at 1:02 PM, Dan DeLeo <notifications@github.com> wrote:

> In package_repo_resource.md
> <https://github.com/chef/chef-rfc/pull/98#discussion_r25796865>:
>
> > +their system.
> > +
> > +## Motivation
> > +
> > +    As a cookbook author,
> > +    I want to write cookbooks that rely only on core resources,
> > +    so that I can minimise my dependencies.
> > +
> > +## Specification
> > +
> > +Create a new `package_repo` resource, similar to
> > +```
> > +package_repo "foo" do
> > +  description "A descriptive string"
> > +  base_url "https://some/url"
> > +  gpgkey "https://some/other/url"
>
> probably everything on Linux will use GPG, but are there any other cases
> which use non-gpg signatures (anyone know how Windows nu-get works)? What
> I'm getting at is, should this just be key ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef-rfc/pull/98/files#r25796865>.
>